pkg/i2gw/providers/common/utils.go

```diff
@@ -200,3 +203,30 @@ func removeBackendRefsDuplicates(backendRefs []gatewayv1.HTTPBackendRef) []gatew
 	}
 	return uniqueBackendRefs
 }
+
+func noNotifications(mType notifications.MessageType, message string, callingObject ...client.Object) {
+}
+
+func mockIngressFromRuleGroup(rg *ingressRuleGroup) client.Object {
```

Comment:
It took me a bit to figure out why we are introducing testing helpers in this package, and then I realized this is not a testing helper, but it's a function to build an Ingress out of a ruleGroup. I suggest renaming it to something like `buildIngressFromRuleGroup` or similar.